File: tests/syn/ruby/01-kitchen.rb
#!/usr/bin/env ruby
# kitchen
require "json"
class Widget
  CONST = 0xff
  def run(arg)
    @value = @@count + $global
    if arg && true
      puts "value #{arg} #@value #{ {inner: {value: arg}} }" %Q{nested {#{arg}} #@value \q}
      puts %q{raw {nested} \\}}, 'it\'s raw'
      puts :plain, :'quoted', :"dynamic #{arg} #@value \q"
      /a\\/b/im =~ arg
    elsif nil
      return 1.5e2ri
    else
      yield self
    end
  rescue Error
    retry
  ensure
    break
  end
end

File: tests/syn/perl/01-kitchen.pl
#!/usr/bin/env perl
# kitchen
use strict;
package Demo;
our $global = 0xff;
my @items = (1, 2.5e3);
sub run {
  local %ENV;
  if ($items[0] eq "x $global\\n") { return q{raw}; }
  elsif ($items[0] ne 'y') { die `false`; }
  else { $items[0] =~ s/a/b/g; }
  $items[0] =~ /word+/im;
  return qq{expanded};
  print `echo $global \q`;
  print 'escaped \\'';
}
